### PR TITLE
feat(forms-web-app): add provide planning application number

### PR DESCRIPTION
**Pull request by a6software**
## Ticket Number

UCD-892

## Description of change

Adds "provide planning application number" form w/ backend integration

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] I have merged commits to avoid fixing things in this PR
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by MrSimonEmms — APPROVED** [review]
lgtm

File: forms-web-app/tests/unit/controllers/appellant-submission/application-number.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        ...req,
+        body: {},
+      };
+      await applicationNumberController.postApplicationNumber(mockRequest, res);
+
+      expect(res.redirect).not.toHaveBeenCalled();
+      expect(logger.error).toHaveBeenCalledWith(error);
+    });
+
+    it('should redirect to `/appellant-submission/supporting-documents` if valid', async () => {
```
Comment (Fenisay) [inline forms-web-app/tests/unit/controllers/appellant-submission/application-number.test.js:55]: should redirect to upload-application

File: forms-web-app/src/views/appellant-submission/application-number.njk [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            isPageHeading: false
+          },
+          id: "application-number",
+          name: "application-number",
+          classes: "govuk-!-width-one-half",
+          spellcheck: false,
+          hint: {
+            text: "You can find this on the decision letter from your local planning department"
+          },
+          value: appeal["application-number"],
```
Comment (Fenisay) [inline forms-web-app/src/views/appellant-submission/application-number.njk:30]: omg mix double quote and single quote...

*REJECT PR*

**Review by Fenisay — APPROVED** [review]

File: forms-web-app/tests/unit/controllers/appellant-submission/application-number.test.js (hunk 1 above)
Comment (a6software) [reply forms-web-app/tests/unit/controllers/appellant-submission/application-number.test.js:55]: whoops yeah, copy paste fail

**Pull request merged**